### ISSUE TITLE
Working with the msg object in external library

### ISSUE THREAD

**Issue by redeemefy** [nextgenhealthcare/connect#19]
Our team wrote/writing a java library to be used within connect. We are running into an issue where I need to pass `msg["some..."]` as a parameter without the `.toString()`. Inside the function I want to determine if the `msg["some..."]` is an `xml` or `json`.

From the library I `logger.warn(":::: JAVA TYPE :::: " + value.javaClass) // org.mozilla.javascript.xmlimpl.XMLList`. Happened that this class comes from a private package.

<img width="840" alt="Screen Shot 2020-03-06 at 8 38 59 AM" src="https://user-images.githubusercontent.com/9165819/76093031-f8579580-5f85-11ea-861e-fdfe3337ad26.png">

Is there a work around or a way to work with the `msg` object in the external library?
In other words, I want to stringify the XML that I'm passing.

**Comment by redeemefy** [comment]
**UPDATE:** 
I'm trying to move this very functionality inside Mirth and still have issues with this. I don't know if this is a bug or not. I'm trying to reach a field within a segment `logger.debug(msg['PID']['PID.4']['PID.4.1'].name());` and I get back `TypeError: name is not a function, it is xml.`. In the `Message Trees` I see that the node is there with an `[empty]` value. I tried to do the same thing with a different field and I do get the node name, for instance `logger.debug(msg['PID']['PID.5']['PID.5.3'].name());` which its value also `[empty]` but I get back `PID.5.3`. 

Why the inconsistency?

**Comment by tonygermano** [comment]
Your actual inbound message probably does not have a PID.4.1 node.

The outbound template in your transformer, if present, gets serialized to the `tmp` variable.

The inbound template is only there for drag-and-drop purposes. `msg` will be the contents of your message, which doesn't necessarily look anything like your inbound template that you view in the `Message Trees` tab.

To your original question, if you are trying to stringify the XML for a node, use `msg['PID']['PID.4']['PID.4.1'].toXMLString()`. That will return an empty string if the node doesn't exist instead of throwing an exception and include the element name even if the node only contains text children.
